includes/class-sensei-lesson.php

```diff
@@ -4726,9 +4731,14 @@ public static function user_not_taking_course_message() {
 	 */
 	public static function course_signup_link() {
 
+		$lesson_id = get_the_ID();
+		if ( $lesson_id ) {
+			return;
+		}
+
 		$course_id = Sensei()->lesson->get_course_id( get_the_ID() );
```

Comment:
```suggestion
		$course_id = Sensei()->lesson->get_course_id( $lesson_id );
```

Very small thing. Just for consistency.